src/Diagnostics.DataProviders/Interfaces/IKustoClient.cs

```diff
@@ -5,6 +5,8 @@ namespace Diagnostics.DataProviders
 {
     public interface IKustoClient
     {
+        Task<DataTable> ExecuteQueryAsync(string query, string cluster, string database, int timeoutSeconds, string requestId = null, string operationName = null);
```

Comment:
Did you decide to stick with the overloaded method name or instead use `ExecuteQueryAsyncForHeartbeat`?

Reply:
I think for this an overload is a better method name. My change in KustoClient.cs one method calls the other just with a default parameter.